controllers/operatorpolicy_controller.go

```diff
@@ -486,6 +598,59 @@ func buildSubscription(
 	return subscription, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCSV(ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	subscription *unstructured.Unstructured,
+) (*operatorv1alpha1.ClusterServiceVersion, error) {
+	// case where subscription is nil
+	if subscription == nil {
+		return nil, nil
+	}
+
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	unstructured := subscription.UnstructuredContent()
+	var sub operatorv1alpha1.Subscription
+
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured, &sub)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the CSV related to the object
+	foundCSV, err := r.DynamicWatcher.Get(watcher, clusterServiceVersionGVK, sub.Namespace,
+		sub.Status.CurrentCSV)
```

Comment:
We probably need to consider the case where this field is not set in the Subscription. It might be similar to the `subscription == nil` case... right after the subscription is created, the status won't have been populated by OLM yet.

Reply:
I'm looking at some other details from my notes, and I think we might want `InstalledCSV` instead. The difference is when there is an InstallPlan pending (for example for an upgrade), I think the `CurrentCSV` points to the "new" version, which won't exist if the plan hasn't been approved yet.